Update the elections process page (#3762)
jenkins-infra/jenkins.io: content/project/board-election-process.adoc

etc.
Such a contribution should be public.

Voter registration is announced through Jenkins mailing lists, blog, and social media accounts.

> Voters can sign up for election using one of two ways:

Reviewer: Optional phrasing change
```suggestion
Voters can register to vote in the election using one of two ways:
```
